<?xml version="1.0"?>
<launch>
  <arg name="exp_rec" default="ltm"/>

  <arg name="debug" default="False"/>
  <arg name="launch-prefix" default="" unless="$(arg debug)"/>
  <arg name="launch-prefix" default="winpdb" if="$(arg debug)"/>

  <!--ltm/mot-->
  <arg name="super_throw" default="false"/>
  <arg name="joystick" default="false"/>
  <arg name="use_robobo" default="false"/>

  <arg name="cal_xmin" default="0.32"/>
  <arg name="cal_xmax" default="1.34"/>
  <arg name="cal_ymin" default="-1.10"/>
  <arg name="cal_ymax" default="1.10"/>
  <arg name="cal_ang" default="0.98125"/>

  <arg name="grab_param_file" default="grab_parameters.yml"/>
  <arg name="throw_param_file" default="throw_parameters.yml"/>
  <arg name="motiven_param_file" default="motiven_parameters.yml"/>
  <arg name="robobo_param_file" default="robobo_parameters.yml"/>

  <arg name="joystick_rate" default="15"/>
  <arg name="joystick_velocity" default="40"/>

  <node name="get_joints_state_server" pkg="mdb_robots_policies" type="get_joints_state" required="true" output="log"/>
  <node name="get_end_state_server" pkg="mdb_robots_policies" type="get_end_state" required="true" output="log"/>
  <node name="get_head_state" pkg="mdb_robots_policies" type="get_head_state" required="true" output="log"/>

  <node name="exp_scene_server" pkg="mdb_robots_policies" type="exp_scene" required="true" output="screen"/>

  <node name="open_grip_left" pkg="mdb_robots_policies" type="open_grip" required="true" output="screen">
    <param name="arm" type="str" value="left" />
  </node>

  <node name="open_grip_right" pkg="mdb_robots_policies" type="open_grip" required="true" output="screen">
    <param name="arm" type="str" value="right" />
  </node>

  <node name="baxter_policies" pkg="mdb_robots_policies" type="baxter_policies" required="true" output="screen" launch-prefix="$(arg launch-prefix)">
    <param name="exp_rec" type="str" value="$(arg exp_rec)" />
    <param name="super_throw" type="bool" value="$(arg super_throw)"/>
    <param name="use_robobo" type="bool" value="$(arg use_robobo)"/>

    <param name="cal_xmin" type="double" value="$(arg cal_xmin)"/>
    <param name="cal_xmax" type="double" value="$(arg cal_xmax)"/>
    <param name="cal_ymin" type="double" value="$(arg cal_ymin)"/>
    <param name="cal_ymax" type="double" value="$(arg cal_ymax)"/>
    <param name="cal_ang" type="double" value="$(arg cal_ang)"/>

    <param name="grab_param_file" type="str" value="$(arg grab_param_file)"/>
    <param name="throw_param_file" type="str" value="$(arg throw_param_file)"/>
    <param name="motiven_param_file" type="str" value="$(arg motiven_param_file)"/>
    <param name="robobo_param_file" type="str" value="$(arg robobo_param_file)"/>
  </node>

  <node name="arduino_joystick_module" pkg="mdb_robots_policies" type="arduino_joystick_module" required="true" output="screen" if="$(arg joystick)"/>
  <node name="robobo_joystick_control" pkg="mdb_robots_policies" type="robobo_joystick_control" required="true" output="screen" if="$(arg joystick)">
    <param name="joystick_rate" type="int" value="$(arg joystick_rate)"/>
    <param name="joystick_velocity" type="int" value="$(arg joystick_velocity)"/>
  </node>

</launch>
